### ISSUE TITLE
Script error on SLES (Suse Linux Enterprise Server) 10 SP4 (64bit)

### ISSUE THREAD

**Issue by LeslieGerman** [drwetter/testssl.sh#999]
Script aborted with 'synax error', see below.
   
1. testssl version from the banner (testssl.sh -b)

```
testssl.sh       2.9.5 from https://testssl.sh/
(0d1e3dfba 2018-02-15 15:15:32 -- )
```

(Even --version or -b fails to run on the SLES.
I obtained the result on RHEL6)

2. what exactly was happening, output is needed
Even run without any option fails:

```
> /share/xpository/github/testssl.sh/2.9.5/dist-bin/testssl.sh
testssl.sh: line 1303: syntax error in conditional expression: unexpected token `|'
testssl.sh: line 1303: syntax error near `|3'
testssl.sh: line 1303: `     [[ -n "$SNI" ]] && [[ " $options " =~ \ -ssl[2|3]\  ]] && options="$(sed "s/$SNI//" <<< "$options")"'
testssl.sh: line 1303: cleanup: command not found
```

3. what did you expect instead?
Run without problem.

4. steps to reproduce

   1. testssl.sh command line: no option.
 
    1. target IP: cannot reveal, sorry
 
    1. openssl version:
      ```
     Using "OpenSSL 1.0.1e-fips 11 Feb 2013" [~102 ciphers]
     on myhost:/usr/bin/openssl
      (built: "Jan 30 07:47:24 EST 2017", platform: "linux-x86_64")
      ```
    
    1. Operating system:
    ```
    > uname -srvmpio
    Linux 2.6.16.60-0.101.1-smp #1 SMP Mon Mar 18 13:32:51 UTC 2013 x86_64 x86_64 x86_64 
    GNU/Linux
    > less /etc/issue
    Welcome to SUSE Linux Enterprise Server 10 SP4  (x86_64)
    ```
    1. bash version:
```
    > bash --version
    GNU bash, version 3.1.17(1)-release (x86_64-suse-linux)

```
(The issue is not critical, actually we can run the script on other OSes. It's ok if you don't fix it.)

**Comment by drwetter** [comment]
thx.

SLES 10 is pretty old. What does Basha --version say?


-- 
Sent from my mobile. Excuse my brevity&typos+the phone's autocorrection

**Comment by drwetter** [comment]
so much for the autocorrection... (sorry)

``bash --version``

**Comment by drwetter** [comment]
ah, mobile=small screen. I am sorry

**Comment by drwetter** [comment]
can you try again please?

With bash 3.2.57 it worked.

**Comment by LeslieGerman** [comment]
Hi Dirk,

Thanks for the fix, I tried it:

```
> ./testssl.sh
./testssl.sh: line 12753: unexpected argument `(' to conditional binary operator
./testssl.sh: line 12753: syntax error near `('
./testssl.sh: line 12753: `     if [[ "$(head -1 "$FNAME")" =~ ( -oG )(.*) ]]; then'

%%./testssl.sh: line 393: : command not found
./testssl.sh: line 393: : command not found
./testssl.sh: line 393: : command not found
./testssl.sh: line 393: : command not found
./testssl.sh: line 625: : command not found
./testssl.sh: line 626: : command not found
```
If you want to make it work on this old SLES, then I can run it for you if you really plan to fix this, but otherwise I think it isn't worth the effort. We can live without it, so don't worry... :)
My bug report basically was just a FYI...

Regards.

**Comment by drwetter** [comment]
Hi @LeslieGerman,

the problem is/was not specifically SLES 10 related but bash. I can live with SLES 10 not being supported anymore by testssl.sh -- as SLES 10 seems not to be supported anymore by Suse (EOL was 2016): https://www.suse.com/lifecycle/

It was new to me that there's a difference between bash 3.1 and bash 3.2.x. To my knowledge the only OS stuck with bash 3 is OSX <sup>1</sup>. That has 3.2.57. 

Bottom line: I appreciate your info & thx. But I'd rather leave it like is then and even maybe revert the previous patch.

Cheers, Dirk

 <sup>1</sup> OpenBSD has no bash at all in their default install and needs to get it from ports

**Comment by LeslieGerman** [comment]
Hi Dirk,

Ahh, ok I see the main point is the bash version differences.

If you leave it as is, that's perfectly OK.
Thanks for your efforts, anyway!

Regards, László
